### ISSUE TITLE
Feature Request: Apps Pane as macOS Dock replacement + keyboard shortcuts

### ISSUE THREAD

**Issue by martinengelke** [Celve/Peninsula#29]
Maybe I got the features wrong, but there are two types of Application/Window switchers supported by Peninsula right now. 

1. CMD-TAB replacement (which is showing all the opened **windows**)
2. Apps (which is also showing the actual opened **apps** _not windows_), which may represent a macOS Dock replacement?

Regarding the second one I would like to have an option to also show minimized/hidden apps. Maybe I could focus the **Peninsula Apps Pane** with a shortcut and then cycling via TAB or SHIFT and pressing the ALT key in addition could reveal all the minimized/hidden apps as well. CMD + Q, CMD + M, CMD + H should work like in the window switcher.

**Comment by Celve** [comment]
Apps actually shows all windows inside a screen. I originally designed it to replace stage manager. 
So you want an app switcher triggered by shortcut `opt + tab`, with the same behavior as macOS standard app switcher?

**Comment by martinengelke** [comment]
> So you want an app switcher triggered by shortcut opt + tab, with the same behavior as macOS standard app switcher?

I don't want anything, it's your app and your choices, I am just asking questions and may point in some direction :p
In the far future it might be good to customize the shortcuts for all the triggers anyway.

My intention was to have a real `macOS Dock` replacement. 
At the moment there are **several approaches** coming along with `Peninsula`.

### Window Switcher
- which shows me `all open windows` across `all open spaces`:
<img width="769" alt="Bildschirmfoto 2024-12-19 um 01 30 41" src="https://github.com/user-attachments/assets/69a8807a-b624-452b-97e6-779593cbc880" />


### Application Panel
- which shows me `all open windows` across `all open spaces` as well, but which I can not access by keyboard, so I have to use my cursor to distinguish between the _single windows_:
<img width="757" alt="Bildschirmfoto 2024-12-19 um 01 31 49" src="https://github.com/user-attachments/assets/6373b068-ea28-449d-9d81-59f7c80f08b9" />


### macOS Dock
- which is also showing `running applications`, including `minimized`, `hidden` and `attached` ones.
<img width="1246" alt="Bildschirmfoto 2024-12-19 um 01 32 54" src="https://github.com/user-attachments/assets/79f07122-dcaf-4391-9c3f-c1c9235ab072" />

---

So, my idea was to let the `Window Switcher` as it actually is, and give the `Application Panel` some extra features, because it seems to be redundant to what the `Window Switcher` is already doing. It _could_ act – that depends on you – more in the direction of the `macOS Dock`, but as you suggested we can also try to imitate `Stage Manager` functionality. 

> I originally designed it to replace stage manager.

**Sounds interesting, maybe you can tell me more about that idea?**

TLDR; I would like to have access to my minimized, hidden and attached applications I can actually only reach with the `macOS Dock` with a simple keystroke plus navigation with TAB and SHIFT plus OPEN, CLOSE, QUIT, MINIMIZE, HIDE. The `macOS Dock` as well as the `Application Panel` of Peninsula are only reachable via mouse control for now.

**Comment by Celve** [comment]
I think it’s a great idea! I’m currently refactoring some code to enable switching between apps and between windows within an app, similar to `Cmd + \`` when stage manager is active. It shouldn’t take too long.

**Comment by Celve** [comment]
For the purpose of Apps, I wanted to design a way for fast switching windows inside a screen, just like what stage manager does (although it also provides stages). So I add Apps to solve it. But now I find it useless because I didn't give it a keyboard shortcuts so basically I won't click the notch for switching.

**Comment by martinengelke** [comment]
>  But now I find it useless because I didn't give it a keyboard shortcuts so basically I won't click the notch for switching.

Exactly what I thought.

> For the purpose of Apps, I wanted to design a way for fast switching windows inside a screen, just like what stage manager does
Maybe not a bad idea at all. Sort of grouping apps together. Highly depends on the UI you will choose for that to not irritate somebody, but could work :)

**Comment by martinengelke** [comment]
> I think it’s a great idea! I’m currently refactoring some code to enable switching between apps and between windows within an app, similar to cmd + ` when stage manager is active. It shouldn’t take too long.

That would be awesome. Uncluttering is always a good idea!

**Comment by martinengelke** [comment]
Another point coming to that section: I would rather like to have a fixed AppIcon size and scroll horizontally instead of having those small icons:
<img width="583" alt="Bildschirmfoto 2024-12-22 um 05 09 13" src="https://github.com/user-attachments/assets/ca51962a-4526-43b7-ace6-659c9d39a4c9" />

Thank you very much!
